src/Templates/src/Microsoft.Maui.Templates.csproj

```diff
@@ -1,5 +1,28 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
+  <UsingTask TaskName="ReplaceFileText" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <InputFilename ParameterType="System.String" Required="true" />
+      <OutputFilename ParameterType="System.String" Required="true" />
+      <MatchExpression ParameterType="System.String" Required="true" />
```

Comment:
what about making this and the replacement parameters into `string[]` so that you don't need to call it repeatedly below, reading and writing the file each time?

Reply:
If an MSBuild experts wants to offer a way to do that, I'm more than happy. I based this off of copying what another build did because I am a copy & paste expert 😁 I agree it's inefficient to re-save multiple times, but in this part of the build it should not add significant time (some low number of milliseconds, presumably).

Reply:
I'm an MSBuild expert 👅 

```xml
<Project>


  <UsingTask TaskName="ReplaceFileText" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
    <ParameterGroup>
      <InputFilenames ParameterType="System.String[]" Required="true" />
      <OutputFilenames ParameterType="System.String[]" Required="true" />
      <MatchExpression ParameterType="System.String" Required="true" />
      <ReplacementText ParameterType="System.String" Required="true" />
    </ParameterGroup>
    <Task>
      <Using Namespace="System" />
      <Using Namespace="System.IO" />
      <Using Namespace="System.Text.RegularExpressions" />
      <Code Type="Fragment" Language="cs">
        <![CDATA[
            if (InputFilenames.Length != OutputFilenames.Length) throw new Exception("..");

            for (int i = 0; i < InputFilenames.Length; i++)
            {
              File.WriteAllText(
                  OutputFilenames[i],
                  Regex.Replace(File.ReadAllText(InputFilenames[i]), MatchExpression, ReplacementText)
                  );
            }
          ]]>
      </Code>
    </Task>
  </UsingTask>

  <Target Name="t">

    <ReplaceFileText
        InputFilenames="a.txt;b.txt"
        OutputFilenames="a2.txt;b2.txt"
        MatchExpression="WIN2D_VERSION_VALUE"
        ReplacementText="BLAH" />

  </Target>
</Project>
```

this works. 😄

Reply:
Oh, ha, thats multiple input/output files, you wanted multiple replacements but the principal is exactly the same.

Reply:
I will try to adapt your C# in XML to modify JSON that updates other XML so that people can use C# to suit my needs 😁

Reply:
ha. Yeah, I'm not sure that the "C# in XML" idea was a great one originally, and I'm guessing that we still don't cache the assemblies either..

Reply:
I adapted it and it seems to work. Thank you for the suggestion!